integrations/src/main/scala/com/quantemplate/integrations/gmaps/GeocodingService.scala

```diff
@@ -0,0 +1,92 @@
+package com.quantemplate.integrations.gmaps
+
+import com.google.maps.{GeocodingApi, GeoApiContext, PendingResult}
+import com.google.maps.errors.ZeroResultsException
+import com.google.maps.model.GeocodingResult
+import scala.concurrent.{Future, Promise}
+import scala.concurrent.ExecutionContext
+import com.google.maps.model.AddressComponent
+import com.google.maps.model.AddressComponentType
+import cats.syntax.traverse.given
+import cats.syntax.option.given
+
+import com.quantemplate.integrations.common.{Config, View, Xlsx}
+
+class GeocodingService()(using conf: Config.GoogleMaps):
+  import GeocodingService.*
+  
+  private lazy val context = GeoApiContext.Builder()
+    .apiKey(conf.apiKey)
+    .build()
+
+
+  def getGeocodedRows(addresses: Vector[String])(using ExecutionContext) =
+    println("getGeocodedRows")
+    addresses
+      .map(geocode)
+      .sequence
+      .map(constructRows)
+   
+  def geocode(address: String): Future[Option[Vector[GeocodingResult]]] = 
```

Comment:
Do we really need `Option[Vector[GeocodingResult]]` here? Cannot be just `Future[Vector[GeocodingResult]]` ?
Do we interpret differently `None` and `Some(Nil)` ?

Reply:
The Google Maps API for a given address could return either an array of results or no results - which is not an error, the latter scenario is a signaled as `Future[None]`

Reply:
But in our use case we treat them in the same way - as empty list?

Reply:
I guess I can use empty lists, but It's a little easier to work with Options here as they are later used to determine the absence of value in specific CSV cells.